vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/network/externalipranger/externalip_admission.go

```diff
@@ -164,8 +181,37 @@ func (r *externalIPRanger) Validate(a admission.Attributes, _ admission.ObjectIn
 			}
 		}
 	}
+
 	if len(errs) > 0 {
-		return apierrs.NewInvalid(a.GetKind().GroupKind(), a.GetName(), errs)
+		//if there are errors reported, resort to RBAC check to see
+		//if this is an admin user who can over-ride the check
+		if !invalidIp {
+			allow, err := r.checkAccess(a)
+			if err != nil {
+				return err
+			}
+			if !allow {
+				return apierrs.NewInvalid(a.GetKind().GroupKind(), a.GetName(), errs)
```

Comment:
as in the restricted endpoints controller, this should be `apierrs.NewForbidden`

Reply:
I only moved the existing error inside the `!allow` block. But for correctness, I can change it to `apierrs.NewForbidden`

Reply:
Actually, the only apierror that takes an error list as argument is `NewInvalid`. Do we want to use `NewForbidden` and only pick the first element in the `errs` slice?

Reply:
It seems like it should have been Forbidden before... anyway, you could aggregate the errors into a single message perhaps